File: tarea1/src/LampControl.java
public class LampControl {

    private Cloud cloud;
    private int channel;

    public LampControl(int channel, Cloud cloud){
        this.channel = channel;
        this.cloud = cloud;
    }

    public void pressPower(){
        cloud.changeLampPowerState(channel);
    }

    public int getChannel(){
        return this.channel;
    }
    
}

File: Tarea_3/Stage3/LampControl.cpp
#include "LampControl.h"

LampControl::LampControl(Cloud* nube)
{
    c = nube;
}

void LampControl::ButtonPressed(int channel)
{
    c->changeLampState(channel);
}

void LampControl::ChangeIntensity(int channel, int value)
{
    c->ChangeIntensityLamp(channel, value);
}
